src/gluonts/model/forecast_generator.py

```diff
@@ -170,7 +170,7 @@ def __call__(
         **kwargs
     ) -> Iterator[Forecast]:
         for batch in inference_data_loader:
-            inputs = [batch[k] for k in input_names]
+            inputs = [batch[k] for k in input_names if k in batch.keys()]
```

Comment:
`input_names` is completely ignored then? Is there a way to update it directly by modifying `get_hybrid_forward_input_names`? I think it is hard to determine this difference (with/out past only features) on-the-fly during runtime?